src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -312,6 +610,35 @@ public IndexList listIndexes() throws PineconeException {
         return indexList;
     }
 
+    /**
+     * Deletes an index.
+     * <p>
+     * Deleting an index is an irreversible operation. All data in the index will be lost.
+     * When you use this command, a request is sent to the Pinecone control plane to delete
+     * the index, but the termination is not synchronous because resources take a few moments to
+     * be released.
+     * <p>
+     * You can check the status of the index by calling the describeIndex command.
+     * With repeated polling of the describeIndex command, you will see the index transition to a
+     * Terminating state before eventually resulting in a 404 after it has been removed.
```

Comment:
Awesome explanation

Reply:
Thanks, but I lifted the copy from @jhamon.